plugins/services/src/js/components/SDKPlans.tsx

```diff
@@ -27,32 +33,65 @@ const getGraphQL = (
       }
     `,
     schema,
-    { serviceName }
+    { serviceId }
   );
 };
 
+const ErrorScreen = () => {
+  return <RequestErrorMsg />;
+};
+
 const LoadingScreen = () => {
   return <Loader />;
 };
 
-const ErrorScreen = () => {
-  return <RequestErrorMsg />;
+const selectedPlan$ = new BehaviorSubject<string>("");
+const handleSelectPlan = (name: string) => {
+  selectedPlan$.next(name);
 };
 
 const SDKPlans = componentFromStream(props$ => {
-  const name$ = (props$ as Observable<{ service: { getId: () => string } }>)
+  const serviceId$ = (props$ as Observable<{
+    service: { getId: () => string };
+  }>)
     .map((props: { service: { getId: () => string } }) => props.service.getId())
     .distinctUntilChanged();
 
-  const plans$ = name$.concatMap((serviceName: string) =>
-    getGraphQL(serviceName).map(
-      (response: { data: { service: Service } }) => response.data.service
+  const plans$ = serviceId$
+    .concatMap((serviceId: string) =>
+      getGraphQL(serviceId).map(
+        (response: { data: { service: Service } }) => response.data.service
```

Comment:
We deconstruct here and then use response, that seems weird

Reply:
That’s a type definition right?

Reply:
I agree with @Poltergeist , I think we are making sure it's a service.

Reply:
Type definition. it's rather easy to mixup those :) We _could_ extract the definition to say    

```ts
type ServiceResponse = { data: { service: Service } };
(response: ServiceResponse) => response.data.service
```